### PR TITLE
Snapshot specific symbol versions

### PR DESCRIPTION
**Pull request by mstampfer**
Added versions parameter to snapshot specific versions of symbols in version store.

### CODE REVIEW

File: arctic/store/version_store.py [hunk 1]
```diff
@@ -666,9 +666,9 @@ def _write_audit(self, user, message, changed_version):
         # Create the audit entry
         mongo_retry(self._audit.insert_one)(audit)
 
-    def snapshot(self, snap_name, metadata=None, skip_symbols=None):
+    def snapshot(self, snap_name, metadata=None, skip_symbols=None, versions=None):
         """
-        Snapshot the current versions of symbols in the library.  Can be used like:
+        Snapshot versions of symbols in the library.  Can be used like:
```
Comment (jamesblackburn) [inline arctic/store/version_store.py]: Would be good to document the form for the `versions` argument

File: arctic/store/version_store.py [hunk 2]
```diff
@@ -688,22 +688,23 @@ def snapshot(self, snap_name, metadata=None, skip_symbols=None):
...
-            skip_symbols = set(skip_symbols)
+        if versions is None:
+            versions = {sym: None for sym in set(self.list_symbols()) - skip_symbols}
 
         # Loop over, and snapshot all versions except those we've been asked to skip
-        for sym in set(self.list_symbols()) - skip_symbols:
+        for sym in versions:
             try:
-                sym = self._read_metadata(sym, read_preference=ReadPreference.PRIMARY)
+                sym = self._read_metadata(sym, read_preference=ReadPreference.PRIMARY, as_of=versions[sym])
```
Comment (jamesblackburn) [inline arctic/store/version_store.py]: Assuming here that every symbol in the library is in the `versions` argument?  What if the library contains additional symbols (or a symbol isn't specified in skip_symbols)?  Perhaps use `get()`?

**Comment by jamesblackburn** [comment]
Would be good to have a test in `test_version_store.py` for the new functionality

File: arctic/store/version_store.py (hunk 2 above)
Comment (richardbounds) [inline arctic/store/version_store.py]: I don't think it is necessary for every symbol to be in `versions` - it is likely to be true, but in our use-case we care about including the symbols that we've passed in.

**Comment by richardbounds** [comment]
The change looks good, but agree it needs a test :)

File: arctic/store/version_store.py (hunk 2 above)
Comment (jamesblackburn) [inline arctic/store/version_store.py]: Ah, sorry I misread the code, I missed the `versions is None` bit...

**Pull request merged**